kedro/config/templated_config.py

```diff
@@ -123,7 +123,7 @@ def __init__(
         self._config_mapping = (
             _get_config_from_patterns(
                 conf_paths=self.conf_paths,
-                patterns=list(globals_pattern),
+                patterns=[globals_pattern],
```

Comment:
So the problem is that if we use `list` it will modify any input like this: `list(*globals.yml)` -> `['*', 'g', 'l', 'o', 'b', 'a', 'l', 's', '.', 'y', 'm', 'l']` whereas if you use `[]` it will just preserve the input like: `[*globals.yml]`

Reply:
The signature for `globals_pattern` is ` globals_pattern: Optional[str] = None` though, so even though lower in the stack we handle lists, up here it's assumed the user only passes in a `str`. See also [this comment](https://github.com/kedro-org/kedro/pull/1438#issuecomment-1103946385)